lifeguardsSilver.java

```diff
@@ -0,0 +1,114 @@
+
+//package usaco2018january;
+//2 hours
+//5,6 timed out, the rest of the tests are a success
+//Difficulty 10/10
+//I was really struggling with this problem and had to look at the solution for some inspiration, I really felt like I learned a lot
+//about silver level programming from this problem

```

Comment:
Good.

Reply:
The problem is difficult due to the intervals overlap. However, if you see the "real" value of any lifeguard is really his *non-overlapping" interval, then it is simply to find out whose "real interval" is minimum.